### PR TITLE
implemented opening connection auth in a child window

### PR DESCRIPTION
**Pull request by a-nahaitsev**
https://trello.com/c/7QO9kx8c

### CODE REVIEW

File: renderer/src/components/JiraConnection/JiraConnection.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const JiraConnection = () => {
     const online = await isOnline();
 
     if (online) {
-      global.ipcRenderer.send(IPC_MAIN_CHANNELS.JIRA_LOGIN);
+      global.ipcRenderer.send("open-child-window", "jira");
```
Comment (okhmelevskyi) [inline renderer/src/components/JiraConnection/JiraConnection.tsx:19]: Could you please move these two strings into a const? electron-src/helpers/constants.ts

File: renderer/src/components/JiraConnection/JiraConnection.tsx [hunk 2]
```diff
@@ -69,19 +76,24 @@ const JiraConnection = () => {
...
   useEffect(() => {
-    if (
-      window.location.search.includes("code") &&
-      window.location.search.includes("state=jiracode") &&
-      !window.location.search.includes("error")
-    ) {
-      (async () => addUser())();
-    }
-  }, []);
+    global.ipcRenderer.on("should-rerender-jira", rerenderListener);
```
Comment (okhmelevskyi) [inline renderer/src/components/JiraConnection/JiraConnection.tsx:91]: The same here.

File: renderer/src/components/JiraConnection/JiraConnection.tsx [hunk 3]
```diff
@@ -69,19 +76,24 @@ const JiraConnection = () => {
...
+
+    return () => {
+      global.ipcRenderer.removeAllListeners("should-rerender-jira");
```
Comment (okhmelevskyi) [inline renderer/src/components/JiraConnection/JiraConnection.tsx:94]: The same here.

File: renderer/src/components/Office365Connection/Office365Connection.tsx [hunk 4]
```diff
@@ -8,15 +8,17 @@ import { LOCAL_STORAGE_VARIABLES } from "@/helpers/contstants";
...
+    ) || []
   );
   const [showEventsInTable, setShowEventsInTable] = useState(false);
 
   const handleSignInButton = async () => {
     const online = await isOnline();
 
     if (online) {
-      global.ipcRenderer.send(IPC_MAIN_CHANNELS.OFFICE365_LOGIN);
+      global.ipcRenderer.send("open-child-window", "office365");
```
Comment (okhmelevskyi) [inline renderer/src/components/Office365Connection/Office365Connection.tsx:21]: The same here.

File: renderer/src/components/Office365Connection/Office365Connection.tsx [hunk 5]
```diff
@@ -91,19 +103,24 @@ const Office365Connection = () => {
...
     ) {
-      (async () => addUser())();
-    }
-
-    if (localStorage.getItem(LOCAL_STORAGE_VARIABLES.SHOW_OFFICE_365_EVENTS) === "true") {
       setShowEventsInTable(true);
     }
-  }, []);
+
+    global.ipcRenderer.on("should-rerender-office365", rerenderListener);
```
Comment (okhmelevskyi) [inline renderer/src/components/Office365Connection/Office365Connection.tsx:118]: The same here.

File: renderer/src/components/Office365Connection/Office365Connection.tsx [hunk 6]
```diff
@@ -91,19 +103,24 @@ const Office365Connection = () => {
...
+
+    return () => {
+      global.ipcRenderer.removeAllListeners("should-rerender-office365");
```
Comment (okhmelevskyi) [inline renderer/src/components/Office365Connection/Office365Connection.tsx:121]: The same here.

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx [hunk 7]
```diff
@@ -18,7 +19,7 @@ const TimetrackerWebsiteConnection = () => {
     const online = await isOnline();
 
     if (online) {
-      global.ipcRenderer.send(IPC_MAIN_CHANNELS.AZURE_LOGIN_BASE);
+      global.ipcRenderer.send("open-child-window", "timetracker-website");
```
Comment (okhmelevskyi) [inline renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx:22]: The same here.

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx [hunk 8]
```diff
@@ -165,16 +175,22 @@ const TimetrackerWebsiteConnection = () => {
...
-      loadUserInfo();
-    } else if (
       searchParams.includes("code") &&
       searchParams.includes("state=azure-additional")
     ) {
       loadPlannerInfo();
     }
+
+    global.ipcRenderer.on(
+      "should-rerender-timetracker-website",
```
Comment (okhmelevskyi) [inline renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx:185]: The same here.

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx [hunk 9]
```diff
@@ -165,16 +175,22 @@ const TimetrackerWebsiteConnection = () => {
...
+      rerenderListener
+    );
+
+    return () => {
+      global.ipcRenderer.removeAllListeners(
+        "should-rerender-timetracker-website"
```
Comment (okhmelevskyi) [inline renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx:191]: The same here.

File: renderer/src/components/TrelloConnection/TrelloConnection.tsx [hunk 10]
```diff
@@ -3,68 +3,66 @@ import { ArrowRightOnRectangleIcon } from "@heroicons/react/24/solid";
...
+
   useEffect(() => {
-    if (
-      window.location.hash.includes("token") &&
-      !window.location.hash.includes("error")
-    ) {
-      (async () => addUser())();
-    }
-  }, []);
+    global.ipcRenderer.on("should-rerender-trello", rerenderListener);
```
Comment (okhmelevskyi) [inline renderer/src/components/TrelloConnection/TrelloConnection.tsx:60]: The same here.

File: renderer/src/components/TrelloConnection/TrelloConnection.tsx [hunk 11]
```diff
@@ -3,68 +3,66 @@ import { ArrowRightOnRectangleIcon } from "@heroicons/react/24/solid";
...
+
+    return () => {
+      global.ipcRenderer.removeAllListeners("should-rerender-trello");
```
Comment (okhmelevskyi) [inline renderer/src/components/TrelloConnection/TrelloConnection.tsx:63]: The same here.

**Review by o-mehlii — APPROVED** [review]

File: renderer/src/components/JiraConnection/JiraConnection.tsx (hunk 2 above)
Comment (a-nahaitsev) [reply renderer/src/components/JiraConnection/JiraConnection.tsx]: resolved

File: renderer/src/components/JiraConnection/JiraConnection.tsx (hunk 3 above)
Comment (a-nahaitsev) [reply renderer/src/components/JiraConnection/JiraConnection.tsx]: resolved

File: renderer/src/components/Office365Connection/Office365Connection.tsx (hunk 4 above)
Comment (a-nahaitsev) [reply renderer/src/components/Office365Connection/Office365Connection.tsx]: resolved

File: renderer/src/components/Office365Connection/Office365Connection.tsx (hunk 6 above)
Comment (a-nahaitsev) [reply renderer/src/components/Office365Connection/Office365Connection.tsx]: resolved

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx (hunk 8 above)
Comment (a-nahaitsev) [reply renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx]: resolved

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx (hunk 9 above)
Comment (a-nahaitsev) [reply renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx]: resolved

File: renderer/src/components/TrelloConnection/TrelloConnection.tsx (hunk 10 above)
Comment (a-nahaitsev) [reply renderer/src/components/TrelloConnection/TrelloConnection.tsx]: resolved

File: renderer/src/components/TrelloConnection/TrelloConnection.tsx (hunk 11 above)
Comment (a-nahaitsev) [reply renderer/src/components/TrelloConnection/TrelloConnection.tsx]: resolved

File: renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx (hunk 7 above)
Comment (a-nahaitsev) [reply renderer/src/components/TimetrackerWebsiteConncetion/TimetrackerWebsiteConncetion.tsx]: resolved

**Pull request closed without merging**